### PR TITLE
Documentation for Scorep

### PR DESCRIPTION
**Pull request by JosephThomasParker**
This adds a Read the Docs page for performance profiling.  At the moment, this is just for Scorep+Scalasa+Cube, with notes on how to set it up on Archer.

### CODE REVIEW

File: manual/sphinx/developer_docs/performance_profiling.rst [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+      return getMesh()->LocalNx;
+    };
+
+**Caution** Instrumenting a function makes it execute more slowly. This can
+result in misleading profiling information, particularly if 
+fast-but-frequently-called functions are instrumented. Try to instrument 
+significant functions only.
+
+This notwithstanding, it is reasonable to expect sensibly-instrumented code to
+run about 50% to 100% slower than the uninstrumented code.
```
Comment (d7919) [inline manual/sphinx/developer_docs/performance_profiling.rst]: I'm not sure about this -- I think if we have this much slow down then it's going to be difficult to really interpret things, I usually find a much smaller reduction in performance when the profiling is limited to things outside of loops over the mesh.

**Comment by bendudson** [comment]
Cool! Thanks @JosephThomasParker

File: manual/sphinx/developer_docs/performance_profiling.rst (hunk 1 above)
Comment (JosephThomasParker) [reply manual/sphinx/developer_docs/performance_profiling.rst]: Ah okay - these figures are for profiling operators on fields.  What would you say is a reasonable overhead for less intensive profiling?

Comment (JosephThomasParker) [reply manual/sphinx/developer_docs/performance_profiling.rst]: Thanks - updated.

File: manual/sphinx/developer_docs/performance_profiling.rst [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+Analysis
+~~~~~~~~
+
+Open the trace file in `Paraver <https://tools.bsc.es/paraver/>`_ with
+
+.. code-block:: bash
+
+    $ wxparaver ${TRACE_NAME}
+
+To view time traces, go to ``File -> Load Congifuration``.  There are many
```
Comment (ZedThree) [inline manual/sphinx/developer_docs/performance_profiling.rst]: Typo: "Congifuration" -> "Configuration"

**Pull request merged**